lightningd/plugin.c

```diff
@@ -630,7 +648,8 @@ static struct io_plan *plugin_write_json(struct io_conn *conn,
 /* This catches the case where their stdout closes (usually they're dead). */
 static void plugin_conn_finish(struct io_conn *conn, struct plugin *plugin)
 {
-	plugin_kill(plugin, "Plugin exited before completing handshake.");
+	const char *msg = tal_fmt(tmpctx, "died %s", state_desc(plugin));
```

Comment:
Very much prefer the `exited` verbiage, since users usually get confused by terms like crash or dying, since they assume it's a fault.